pgns/127489.js

```diff
@@ -136,3 +138,156 @@ module.exports = [
   }
 ]
+
+var status1Notifications = [
+  {
+    node: 'notifications.propulsion.%s.checkEngine',
+    message: 'Check %s Engine'
+  },
+  {
+    node: 'notifications.propulsion.%s.overTemperature',
+    message: '%s Engine Over Temperature'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowOilPressure',
+    message: '%s Engine Low Oil Pressure'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowOilLevel',
+    message: '%s Engine Low Oil Level'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowFuelPressure',
+    message: '%s Engine Low Fuel Pressure'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowSystemVoltage',
+    message: '%s Low System Voltage'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowCoolantLevel',
+    message: '%s Engine Low Coolant Level'
+  },
+  {
+    node: 'notifications.propulsion.%s.waterFlow',
+    message: '%s Engine Water Flow'
+  },
+  {
+    node: 'notifications.propulsion.%s.waterInFuel',
+    message: '%s Water in Fuel'
+  },
+  {
+    node: 'notifications.propulsion.%s.chargeIndicator',
+    message: '%s Engine Charge Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.preheatIndicator',
+    message: '%s Preheat Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.highBoostPressure',
+    message: '%s Engine High Boost Pressure'
+  },
+  {
+    node: 'notifications.propulsion.%s.revLimitExceeded',
+    message: '%s Engine Rev Limit Exceeded'
+  },
+  {
+    node: 'notifications.propulsion.%s.eGRSystem',
+    message: '%s Engine EGR System'
+  },
+  {
+    node: 'notifications.propulsion.%s.throttlePositionSensor',
+    message: '%s Engine Throttle Position Sensor'
+  },
+  {
+    node: 'notifications.propulsion.%s.emergencyStopMode',
+    message: '%s Engine Emergency Stop Mode'
+  }
+];
+
+var status2Notifications = [
+  { 
+    node: 'notifications.propulsion.%s.warningLevel1',
+    message: '%s Engine Warning Level 1'
+  },
+  {
+    node: 'notifications.propulsion.%s.warningLevel2',
+    message: '%s Engine Warning Level 2'
+  },
+  {
+    node: 'notifications.propulsion.%s.powerReduction',
+    message: '%s Engine Power Reduction'
+  },
+  {
+    node: 'notifications.propulsion.%s.maintenanceNeeded',
+    message: '%s Engine Maintenance Needed'
+  },
+  {
+    node: 'notifications.propulsion.%s.commError',
+    message: '%s Engine Comm Error'
+  },
+  {
+    node: 'notifications.propulsion.%s.subOrSecondaryThrottle',
+    message: '%s Engine Sub or Secondary Throttle'
+  },
+  {
+    node: 'notifications.propulsion.%s.neutralStartProtect',
+    message: '%s Neutral Start Protect'
+  },
+  {
+    node: 'notifications.propulsion.%s.shuttingDown',
+    message: '%s Engine Shutting Down'
+  }
+];
+
+
+function generateMappingsForStatus(field, notifications)
+{
+  notifications.forEach((notif,index) => {
+    var mapping = {
+      node: function(n2k) {
+        var engine = n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port' ? "port" : "starboard";
```

Comment:
The case is intentional. One is for the notification path `notification.engine.port`. The other is for the message "Check Port Engine"